File: src/ruby/lib/dependabot/job/models/enums/requirements_update_strategy.rb
# typed: strict
# frozen_string_literal: true

require 'sorbet-runtime'

module Dependabot
  module Job
    module Models
      module Enums
        # Represents the strategy for updating dependency requirements.
        class RequirementsUpdateStrategy < T::Enum
          enums do
            # Automatically determine the best strategy.
            Auto = new('auto')

            # Widen version ranges when possible.
            Widen = new('widen')

            # Increase version requirements to minimum supported.
            Increase = new('increase')

            # Increase version requirements only if required.
            IncreaseIfNecessary = new('increase_if_necessary')

            # Lock to exact version.
            LockfileOnly = new('lockfile_only')

            # Bump versions according to requirements.
            BumpVersions = new('bump_versions')
          end
        end
      end
    end
  end
end

File: src/dotnet/src/Dependabot.Job/Models/Enums/RequirementsUpdateStrategy.cs
using System.Text.Json.Serialization;

namespace Dependabot.Job.Enums;

/// <summary>
/// Represents the strategy for updating dependency requirements.
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter<RequirementsUpdateStrategy>))]
public enum RequirementsUpdateStrategy
{
  /// <summary>
  /// Automatically determine the best strategy.
  /// </summary>
  Auto,

  /// <summary>
  /// Widen version ranges when possible.
  /// </summary>
  Widen,

  /// <summary>
  /// Increase version requirements to minimum supported.
  /// </summary>
  Increase,

  /// <summary>
  /// Increase version requirements only if required.
  /// </summary>
  IncreaseIfNecessary,

  /// <summary>
  /// Lock to exact version.
  /// </summary>
  LockfileOnly
}
